src/test/java/com/arpnetworking/metrics/common/integration/KafkaIT.java

```diff
@@ -0,0 +1,253 @@
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Tests to check integration with a kafka topic.
+ *
+ * @author Joey Jackson (jjackson at dropbox dot com)
+ */
+public class KafkaIT {
+    private static final String KAFKA_SERVER = "localhost:9092";
+    private static final String KEY_SERIALIZER = "org.apache.kafka.common.serialization.IntegerSerializer";
+    private static final String VALUE_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+    private static final String KEY_DESERIALIZER = "org.apache.kafka.common.serialization.IntegerDeserializer";
+    private static final String VALUE_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+    private static final int POLL_DURATION_MILLIS = 1000;
+    private static final int TIMEOUT = 5000;
+
+    private Map<String, Object> _consumerProps;
+    private KafkaConsumer<Integer, String> _consumer;
+    private String _topicName;
+    private KafkaSource<String, String> _source;
+    private List<ProducerRecord<Integer, String>> _producerRecords;
+
+    @Before
+    public void setUp() {
+        // Create kafka topic
+        _topicName = createTopicName();
+        createTopic(_topicName);
+
+        // Create and send producer records
+        _producerRecords = createProducerRecords(_topicName, 3);
+        sendRecords(_producerRecords);
+
+        // Create consumer props
+        _consumerProps = Maps.newHashMap();
+        _consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA_SERVER);
+        _consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer0");
+        _consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, KEY_DESERIALIZER);
+        _consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, VALUE_DESERIALIZER);
+        _consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        _consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "group" + _topicName);
+    }
+
+    @After
+    public void tearDown() {
+        if (_source != null) {
+            _source.stop();
+        }
+    }
+
+    @Test
+    public void testKafkaSourceSingleObserver() {
+        // Create Kafka Source
+        _consumer = new KafkaConsumer<>(_consumerProps);
+        _consumer.subscribe(Collections.singletonList(_topicName));
+        _source = new KafkaSource.Builder<String, String>()
+                .setName("KafkaSource")
+                .setParser(new StringParser.Builder().build())
+                .setConsumer(_consumer)
+                .setPollTimeMillis(POLL_DURATION_MILLIS)
+                .build();
+
+        // Observe records
+        final Observer observer = Mockito.mock(Observer.class);
+        _source.attach(observer);
+        _source.start();
+        for (ProducerRecord<Integer, String> expected : _producerRecords) {
+            Mockito.verify(observer, Mockito.timeout(TIMEOUT)).notify(_source, expected.value());
+        }
```

Comment:
Out of paranoia, could we verify that those were the _only_ messages processed, and that they were each processed exactly once? I think `Mockito.verify(observer, Mockito.times(_producerRecords.size())).notify(_source, Mockito.any())` would do it.